src/test/java/it/gov/pagopa/arc/service/bizevents/BizEventsServiceImplTest.java

```diff
@@ -206,5 +229,78 @@ void givenTransactionIdWhenCallRetrieveTransactionReceiptFromBizEventsThenReturn
+        Mockito.verify(bizEventsPaidNoticeDTO2NoticeDTOMapperMock).toNoticeDTOList(any());
+        Mockito.verify(bizEventsPaidNoticeListDTO2NoticesListDTOMapperMock).toNoticesListDTO(anyList());
+        Mockito.verify(bizEventsPaidResponseDTO2NoticesListResponseDTOMapperMock).toNoticesListResponseDTO(any(), anyString());
+
+    }
+
+    @Test
+    void givenParametersWhenCalRetrievePaidListFromBizEventsThenReturnNull() {
+        //given
+        when(bizEventsPaidNoticeConnectorMock.getPaidNoticeList(DUMMY_FISCAL_CODE,CONTINUATION_TOKEN,SIZE, true, true, ORDER_BY, ORDERING)).thenReturn(null);
+        //when
+        NoticesListResponseDTO result = bizEventsService.retrievePaidListFromBizEvents(CONTINUATION_TOKEN, SIZE, true, true, ORDER_BY, ORDERING);
+
+        //then
+        Assertions.assertNotNull(result);
+        Assertions.assertEquals(Collections.emptyList(), result.getNoticesListDTO().getNotices());
+        Assertions.assertNull(result.getContinuationToken());
+
+        Mockito.verify(bizEventsPaidNoticeConnectorMock).getPaidNoticeList(anyString(),anyString(),anyInt(),anyBoolean(),anyBoolean(),anyString(),anyString());
+        Mockito.verify(bizEventsPaidNoticeDTO2NoticeDTOMapperMock, never()).toNoticeDTOList(any());
+        Mockito.verify(bizEventsPaidNoticeListDTO2NoticesListDTOMapperMock, never()).toNoticesListDTO(anyList());
+        Mockito.verify(bizEventsPaidResponseDTO2NoticesListResponseDTOMapperMock, never()).toNoticesListResponseDTO(any(), anyString());
+    }
+
+    @Test
+    void givenParametersWhenCalRetrievePaidListFromBizEventsThenReturnEmptyResult() {
+        //given
+        BizEventsPaidResponseDTO bizEventsPaidResponseDTO = BizEventsPaidResponseDTO.builder().notices(new ArrayList<>()).continuationToken("").build();
+
+        when(bizEventsPaidNoticeConnectorMock.getPaidNoticeList(DUMMY_FISCAL_CODE,CONTINUATION_TOKEN,SIZE, true, true, ORDER_BY, ORDERING)).thenReturn(bizEventsPaidResponseDTO);
+        //when
+        NoticesListResponseDTO result = bizEventsService.retrievePaidListFromBizEvents(CONTINUATION_TOKEN, SIZE, true, true, ORDER_BY, ORDERING);
+
+        //then
+        Assertions.assertNotNull(result);
+        Assertions.assertEquals(Collections.emptyList(), result.getNoticesListDTO().getNotices());
+        Assertions.assertNull(result.getContinuationToken());
+
+        Mockito.verify(bizEventsPaidNoticeConnectorMock).getPaidNoticeList(anyString(),anyString(),anyInt(),anyBoolean(),anyBoolean(),anyString(),anyString());
+        Mockito.verify(bizEventsPaidNoticeDTO2NoticeDTOMapperMock, never()).toNoticeDTOList(any());
+        Mockito.verify(bizEventsPaidNoticeListDTO2NoticesListDTOMapperMock, never()).toNoticesListDTO(anyList());
+        Mockito.verify(bizEventsPaidResponseDTO2NoticesListResponseDTOMapperMock, never()).toNoticesListResponseDTO(any(), anyString());
```

Comment:
instead of `verify` `never()`, add an `AfterEach` method where to call `Mockito.verifyNotMoreInteraction` on all mocks.
this will introduce a check where for each test it will throw an error if there is not a verified interaction (a `when` clause with conditions is also a verify)

Reply:
But @AfterEach Method with verifyNoInteractions(); will be used for all methods and it is not correct, I would only check in this method if there are no interactions with Mock.